docflow/bot/extract/cpp.py

```diff
@@ -0,0 +1,103 @@
+"""
+Contains extractor functions for
+scraped data from cppreference.com.
+"""
+
+from typing import Optional
+
+import discord
+
+from .cpp_embed import CppEmbed
+from .search import search
+from .util import get_ref
+
+CPP_STUBS = get_ref("cpp_stubs.json")
+CPP_SYMBOLS = get_ref("cpp_symbols.json")
+
+
+def stub(query: str) -> Optional[discord.Embed]:
+    """
+    Searches for the given query in the
+    C++ stub database, for example
+    "Strings Library".
+    """
+
+    names = [obj['name'] for obj in CPP_STUBS]
+    search_result = search(names, query)
+    for stub_obj in CPP_STUBS:
+        if search_result == stub_obj['name']:
+            stub_ = stub_obj
+            break
+    else:
+        return None
+
+    embed = CppEmbed(stub_)
+    for header in stub_['items']:
+        embed.add_field(
+            name=header,
+            value=stub_['items'][header].strip() or "Nothing found here :("
+        )
+    return embed
+
+
+def symbol(name: str) -> Optional[discord.Embed]:
+    """
+    Extracts the given C++ symbol from the
+    C++ symbol index, for example std::cout.
+    """
+
+    def parse_function(symb: dict):
+        """Parses a function symbol, such as std::abs."""
+
+        response.add_field(
+            name="Parameters",
+            value='\n'.join(symb['params']) or "No parameters found."
+        ).add_field(
+            name="Return Value",
+            value=symb['return'] or "No non-garbage return value found :("
+        )
+
+    def parse_type(symb: dict):
+        """Parses a type symbol, such as std::vector."""
+
+        def member_string(member_dict):
```

Comment:
From what I know some guideline says that all non-anonymous functions should be denoted as `def xyz` instead of `xyz = lambda`. But @aeshthetic wrote this.